lib/Doctrine/DBAL/Schema/MySqlSchemaManager.php

```diff
@@ -284,4 +285,38 @@ protected function _getPortableTableForeignKeysList($tableForeignKeys)
 
         return $result;
     }
+
+    public function listTableDetails($tableName)
+    {
+        $table = parent::listTableDetails($tableName);
+
+        /** @var MySqlPlatform $platform */
+        $platform = $this->_platform;
+        $sql      = $platform->getListTableMetadataSQL($tableName);
+
+        $tableOptions = $this->_conn->fetchAssoc($sql);
+
+        $table->addOption('engine', $tableOptions['ENGINE']);
+        $table->addOption('collation', $tableOptions['TABLE_COLLATION']);
+        if ($tableOptions['AUTO_INCREMENT'] !== '') {
+            $table->addOption('autoincrement', $tableOptions['AUTO_INCREMENT']);
+        }
+        if ($tableOptions['TABLE_COMMENT'] !== '') {
+            $table->addOption('comment', $tableOptions['TABLE_COMMENT']);
+        }
+        if ($tableOptions['CREATE_OPTIONS'] === '') {
+            return $table;
+        }
+
+        $createOptions = [];
+        foreach (explode(' ', $tableOptions['CREATE_OPTIONS']) as $option) {
```

Comment:
It won't work with an empty value of `$tableOptions['CREATE_OPTIONS']`. Please have it covered with a test.